### ISSUE TITLE
Unnecessary references for casting from immutable reference to mutable reference.

### ISSUE THREAD

**Issue by ticki** [rust-lang/rust-clippy#1212]
A warning should be generated with:

``` rust
let mut a = 2;
let mut a_mut = &mut a;
let a_ref: &u8 = &a_mut;
```

which is more approriately written as:

``` rust
let mut a = 2;
let mut a_mut = &mut a;
let a_ref: &u8 = a_mut;
```

(when possible)

**Comment by mcarton** [comment]
I would have guessed this was already covered by [`needless_borrow`](https://manishearth.github.io/rust-clippy/current/#needless_borrow), but it seems like not.

**Comment by oli-obk** [comment]
I think the issue is that this is a conversion from a `&&mut T` to a `&T`. I'll check how we can improve the `needless_borrow` lint.
